cmd/snap/cmd_run.go

```diff
@@ -49,25 +52,52 @@ func init() {
 }
 
 func (x *cmdRun) Execute(args []string) error {
-	return snapRun(x.Positional.SnapApp, x.Command, args)
-}
-
-func getSnapInfo(snapName string) (*snap.Info, error) {
-	// we need to get the revision here because once we are inside
-	// the confinement the snapd API may be unavailable.
-	snaps, err := Client().List([]string{snapName})
-	if err != nil {
-		return nil, err
+	// Catch some invalid parameter combinations, provide helpful errors
+	if x.Hook != "" && x.Command != "" {
+		return fmt.Errorf("invalid parameters: --hook cannot be used with --command")
 	}
-	if len(snaps) == 0 {
-		return nil, fmt.Errorf("cannot find snap %q", snapName)
+	if x.Revision != "" && x.Hook == "" {
+		return fmt.Errorf("invalid parameters: --revision can only be used with --hook")
+	}
+	if x.Hook != "" && len(args) > 0 {
+		return fmt.Errorf("invalid parameters: extra arguments cannot be used when using --hook")
```

Comment:
What about:

`fmt.Errorf("too many arguments for hook %s: %s", x.Hook, args[0])`

?

It's generally handy to see what was in the command line that shouldn't be there.